### PR TITLE
Introduce Try#getOrElse((Throwable)->B) and rename the old impl getOrDefault

### PR DESCRIPTION
**Pull request by kevinherron**
These extension functions can't exist under the same name because of type erasure.

@pakoito suggested in Slack that renaming the existing signature to `getOrDefault` and using `getOrElse` for the new signature was okay.

### CODE REVIEW

File: kategory-docs/docs/docs/datatypes/try/README.md [hunk 1]
```diff
@@ -63,12 +63,26 @@ val lotteryTry = Try { getLotteryNumbers() }
...
+lotteryTry.getOrElse { ex: Throwable -> emptyList() }
+```
+
+`getOrElse` can generally be used anywhere `getOrDefault` is used, ignoring the exception if it's not needed:
 
 ```kotlin:ank
 lotteryTry.getOrElse { emptyList() }
 ```
 
+
```
Comment (pakoito) [inline kategory-docs/docs/docs/datatypes/try/README.md]: Unnecessary additional lines

**Comment by pakoito** [comment]
We cannot currently merge as @raulraja is going through a large refactor and we'll need to integrate it later!

**Comment by raulraja** [comment]
@ersin-ertan can you update your PR with latest from master so we can merge it? thanks!

**Comment by kevinherron** [comment]
Updated 👍

**Pull request merged**